Drivers/Src/CAN.c

```diff
@@ -0,0 +1,120 @@
+#include "CAN.h"
+
+static void floatToBytes(float val, uint8_t* bytes_array, uint8_t bytes);
+
+/**
+ * @brief   Initializes the CAN system
+ * @param   None
+ * @return  None
+ */
+
+// Not sure if we want CAN1 to be an argument to CAN_Init or not
+void CAN_Init(void) {
+    BSP_CAN_Init(CAN1);
+}
+
+/**
+ * @brief   Transmits data onto the CANbus
+ * @param   id : CAN id of the message
+ * @param 	payload : the data that will be sent.
+ * @return  0 if data wasn't sent, otherwise it was sent.
+ */
+int CANbus_Send(CANId_t id, CANPayload_t payload) {
+
+	uint8_t bytes = 4; // Not sure how we want to decide the number of wanted bytes. An extra paramter to this function?
+	
+
+	// Not sure if these are the ids we want but I took them from BPS as placeholders
+	switch (id) {
+		case TRIP:
+			return BSP_CAN_Write(id, &payload.data.b, 1);
+
+		case ALL_CLEAR:
+			return BSP_CAN_Write(id, &payload.data.b, 1);
+
+		case CONTACTOR_STATE:
+			return BSP_CAN_Write(id, &payload.data.b, 1);
+
+		case CURRENT_DATA:
+			uint8_t txdata[bytes];
+			floatToBytes(payload.data.f, &txdata[0]);
+			return BSP_CAN_Write(id, txdata, bytes);
+
+		case VOLT_DATA:
+		case TEMP_DATA:
+			uint8_t txdata[bytes+1];
+			txdata[0] = payload.idx;
+			floatToBytes(payload.data.f, &txdata[1]);
+			return BSP_CAN_Write(id, txdata, bytes);
+
+		case SOC_DATA:
+			uint8_t txdata[bytes];
+			floatToBytes(payload.data.f, &txdata[0]);
+			return BSP_CAN_Write(id, txdata, bytes);
+
+		case WDOG_TRIGGERED:
+			return BSP_CAN_Write(id, &payload.data.b, 1);
+
+		case CAN_ERROR:
+			return BSP_CAN_Write(id, &payload.data.b, 1);

```

Comment:
Same thing as the header file with the IDs. Since this is for the send function, we only need it to work for the Car State command and all the Motor Controller information commands right now.